src/platform/unix/mod.rs

```diff
@@ -962,6 +960,20 @@ fn create_shmem(name: CString, length: usize) -> c_int {
     }
 }
 
+#[cfg(target_os="freebsd")]
+fn create_shmem(_name: CString, length: usize) -> c_int {
+    unsafe {
+        // NB: the FreeBSD man page for shm_unlink states that it requires
+        // write permissions, but testing shows that read-write is required.
+        let fd = libc::shm_open(1 as *mut i8, // SHM_ANON
```

Comment:
:+1: Could we add this to `libc`

Reply:
Sure — https://github.com/rust-lang/libc/pull/837

Reply:
What's the status on this? Could you perhaps request a new release to get this functionality exposed?...